### DISCUSSION TITLE
Exporting custom Mantine library for next.js

### DISCUSSION THREAD

**Discussion by blake-epistemix** [mantinedev/mantine#4128] [Q&A]
We've created a component library powered by Mantine that we've used for several projects, but we're trying to create a next.js app that uses these components and we're having issues with next.js "server" rendering when working on the project locally. Our library uses Vite to build our custom components and styles as well as exporting Mantine's components. However, when importing any Mantine component into our next.js project from our library, we get "ReferenceError: document is not defined" errors.

Has anyone attempted something similar and know a work around? Is it a vite, next, or mantine problem? We've tried the bare minimum of just having our library with the line `export * from @mantine/core`. Then in our next.js app, we'll try `import { MantineProvider } from 'our-library'` and it gives us the document error, but if we switch to import directly from @mantine/core, then it works.

**Comment by rtivital** [comment]
The issue is with your bundler – you can use rollup or webpack to bundle packages

**Reply by blake-epistemix** [reply]
Vite uses rollup for it's library mode which is what I'm using. It provides rollup options, so if it needs specific settings, I should be able to modify it.
